src/pytest_netdut/wrappers.py

```diff
@@ -34,36 +34,86 @@ def _splitcmds(cmds):
-        (r"l1 source interface (.*)", r"source \1"),
-        (
-            r"l1 source mac",
-            r"source mac",
-        ),
-        (r"no l1 source", r"no source"),
-        (r"bash sudo cortina", "CAN NOT TRANSLATE"),
-        (r"traffic-loopback source network device phy", r"loopback internal"),
-        (r"traffic-loopback source system device phy", r"loopback"),
-        (r"no traffic-loopback", r"no loopback"),
-    ]
-
-    commands = []
-    for command in _splitcmds(cmds):
-        for (before, after) in translations:
-            matcher = re.match(before, command)
-            if matcher:
-                commands.append(matcher.expand(after))
-                break
-        else:
-            # If we don't get a match, just append the original.
-            commands.append(command)
-    if commands != cmds:
-        logging.debug("Before: %s, After: %s", repr(cmds), repr(commands))
-    return commands
+class Translator:
+    """Base class which implements translation functions for hiding syntax
+    and result differences between devices with different operating systems.
+
+    Override the standardize_key function to implement result processing.
+
+    Args:
+        config_patterns (iterable[regex pattern]): List of regex patterns for CLI translations.
+    """
+
+    def __init__(self, config_patterns):
+        self.config_patterns = config_patterns
+
+    def standardize_key(self, k):
+        """Returns a standardized key.
+
+        Args:
+            k (str): Key to process.
+        """
+        raise NotImplementedError
+
+    def process_dict(self, d):
+        result = {}
+        for k, v in d.items():
+            nk = self.standardize_key(str(k))
+            result[nk] = self.process_dict(v) if isinstance(v, dict) else v
+        return result
+
+    def cli(self, cmds):
+        """CLI translator.
+
+        Args:
+            cmds (str | Iterable[str]): Either an iterable of commands to be run or a newline separated string containing one or more commands.
+        """
+        commands = []
+        for command in _splitcmds(cmds):
+            for (before, after) in self.config_patterns:
+                matcher = re.match(before, command)
+                if matcher:
+                    commands.append(matcher.expand(after))
+                    break
+            else:
+                # If we don't get a match, just append the original.
+                commands.append(command)
+        if commands != cmds:
+            logging.debug("Before: %s, After: %s", repr(cmds), repr(commands))
+        return commands
+
+    def json(self, data):
+        """JSON translator.
+
+        Args:
+            data (str | dict): Results are typically a string or a dictionary.
+        """
+        if isinstance(data, dict):
+            return self.process_dict(data)
+        return data
+
+
+class EOS_MOS_Translator(Translator):
+    """Standardize config and results to EOS."""
+
+    def __init__(self):
```

Comment:
I think this should have a compatible signature with the base class.

Do we want `config_patterns` to be an argument, or is it always a static feature of the class?